internal/git/util.go

```diff
@@ -40,7 +39,12 @@ func CloneRepo(
 	if err != nil {
 		return nil, fmt.Errorf("error getting  the  clone options: %v", err)
 	}
-	return cloneRepo(ctx, gitSync, cloneOptions)
+
+	fetchOptions, err := gitShared.GetRepoFetchOptions(ctx, client, gitSync.Spec.RepoUrl, globalConfig)
```

Comment:
nit: can this pass in the `gitCredentials` rather than the `globalConfig` for 1: symmetry; 2. to avoid the double call to `FindCredByUrl()`?